### PR TITLE
Columns api

### PR DESCRIPTION
**Pull request by Keerthi-212839**

### CODE REVIEW

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java [hunk 1]
```diff
@@ -11,6 +11,7 @@
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import com.ust.sourcecourse.configuration.entity.Project;
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:14]: Compilation is failing. Do not added configuration service as dependency in this orch service. 
Remove this import and in service classes as well.

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java [hunk 2]
```diff
@@ -1,16 +1,28 @@
...
 import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 import org.springframework.web.client.RestTemplate;
+import org.springframework.web.server.ResponseStatusException;
 
+import com.ust.sourcecourse.configuration.entity.Project;
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java:20]: Compilation is failing. Do not added configuration service as dependency in this orch service. 
Do not ad entities & repository of other services here.

**Review by shijukbabu-ust — CHANGES_REQUESTED** [review]

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 1 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java (hunk 2 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java]: resolved

Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 1 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java [hunk 3]
```diff
@@ -1,16 +1,26 @@
...
 import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 import org.springframework.web.client.RestTemplate;
-
+import org.springframework.web.server.ResponseStatusException;
+import com.ust.sourcecourse.configuration.entity.ProjectGroup;
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java:19]: Remove entity and repository imports

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java [hunk 4]
```diff
@@ -4,17 +4,18 @@
...
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
-
 import com.ust.sourcecourse.orchestrator.request.ProjectGroupRequest;
 import com.ust.sourcecourse.orchestrator.response.ProjectGroupResponse;
 import com.ust.sourcecourse.orchestrator.service.ProjectGroupService;
 @RestController
+@CrossOrigin(origins = "*", allowedHeaders = "*")
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:18]: Add this line in ProjectController class also.

**Review by shijukbabu-ust — CHANGES_REQUESTED** [review]

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java (hunk 3 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 4 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:22]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java [hunk 5]
```diff
@@ -4,29 +4,58 @@
...
-    private final ProjectGroupService projectGroupService;
+	private final ProjectGroupService projectGroupService;
+
+	@Autowired
+	public ProjectGroupController(ProjectGroupService projectGroupService) {
+		this.projectGroupService = projectGroupService;
+	}
+
+	@PostMapping
+	public ResponseEntity<List<ProjectGroupResponse>> createProjectGroup(
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:34]: Check latest code in config service. It returns of type ProjectGroupResponse and not List<ProjectGroupResponse>

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java [hunk 6]
```diff
@@ -4,29 +4,58 @@
...
+			@RequestBody ProjectGroupRequest projectGroupRequest) {
+		return projectGroupService.createProGroup(projectGroupRequest);
+	}
+
+	@GetMapping
+	public ResponseEntity<List<ProjectGroupResponse>> getProjectGroupsByProjectUid(@PathVariable Long uid) {
+		List<ProjectGroupResponse> projectGroups = (List<ProjectGroupResponse>) projectGroupService
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:41]: The service return type is ResponseEntity<List<ProjectGroupResponse>>. Take as that type itself. Do not typecast.

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java [hunk 7]
```diff
@@ -4,29 +4,58 @@
...
+				.findByProjectUid(uid);
+		if (!projectGroups.isEmpty()) {
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:43]: This check is already done at Service class right? So no need to check here again. Directly return the result.

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java [hunk 8]
```diff
@@ -19,24 +23,74 @@ public class ProjectGroupService {
...
         String url = configServiceURL + "/project-groups";
         HttpHeaders headers = new HttpHeaders();
         headers.setContentType(MediaType.APPLICATION_JSON);
         HttpEntity<ProjectGroupRequest> request = new HttpEntity<>(projectGroupRequest, headers);
         
-        ResponseEntity<ProjectGroupResponse> response = restTemplate.exchange(
-            url, HttpMethod.POST, request, ProjectGroupResponse.class);
-        
+        ResponseEntity<List<ProjectGroupResponse> >response = restTemplate.exchange(
+            url, HttpMethod.POST, request, new ParameterizedTypeReference<List<ProjectGroupResponse>>() {
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java:39]: It should not be list.

**Review by shijukbabu-ust — CHANGES_REQUESTED** [review]

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 1 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java (hunk 2 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java (hunk 3 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 4 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:22]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 5 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 6 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 7 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java [hunk 9]
```diff
@@ -4,29 +4,58 @@
...
+		this.projectGroupService = projectGroupService;
+	}
+
+	@PostMapping
+	public ResponseEntity<List<ProjectGroupResponse>> createProjectGroup(
+			@RequestBody ProjectGroupRequest projectGroupRequest) {
+		return projectGroupService.createProGroup(projectGroupRequest);
+	}
+
+	@GetMapping
```
[the remark this answers is not in the document]
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java [hunk 10]
```diff
@@ -4,29 +4,58 @@
...
+		List<ProjectGroupResponse> projectGroups = (List<ProjectGroupResponse>) projectGroupService
+				.findByProjectUid(uid);
+		if (!projectGroups.isEmpty()) {
+			return ResponseEntity.ok(projectGroups);
+		} else {
+			return ResponseEntity.noContent().build();
+		}
+	}
+
+	@PutMapping()
```
[the remark this answers is not in the document]
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:45]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java [hunk 11]
```diff
@@ -4,29 +4,58 @@
...
-        this.projectGroupService = projectGroupService;
-    }
+	}
 
-    @PostMapping
-    public ResponseEntity<ProjectGroupResponse> createProjectGroup(@RequestBody ProjectGroupRequest projectGroupRequest) {
-        return projectGroupService.createProGroup(projectGroupRequest);
-    }
-   }
+	@DeleteMapping
```
[the remark this answers is not in the document]
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:52]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java (hunk 8 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java [hunk 12]
```diff
@@ -19,24 +23,74 @@ public class ProjectGroupService {
...
+            });
+      
         return response;
     }
-}
+    public ResponseEntity<List<ProjectGroupResponse>> findByProjectUid(Long id) {
+    	
+    	String url = configServiceURL + "/project-groups/" + id  ;
```
[the remark this answers is not in the document]
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java [hunk 13]
```diff
@@ -19,24 +23,72 @@ public class ProjectGroupService {
...
+    
     public ProjectGroupService(@Value("${config.service.url}") String configServiceURL, RestTemplate restTemplate) {
         this.configServiceURL = configServiceURL;
         this.restTemplate = restTemplate;
     }
-
-    public ResponseEntity<ProjectGroupResponse> createProGroup(@Valid ProjectGroupRequest projectGroupRequest) {
-        String url = configServiceURL + "/project-groups";
+    public ResponseEntity<ProjectGroupResponse> createProjectGroup(@Valid ProjectGroupRequest projectGroupRequest) {
+        String url = configServiceURL + "Project group with uid ";
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java:32]: Why is this changed? It should be /project-groups

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java [hunk 14]
```diff
@@ -19,24 +23,72 @@ public class ProjectGroupService {
...
         HttpHeaders headers = new HttpHeaders();
         headers.setContentType(MediaType.APPLICATION_JSON);
-        HttpEntity<ProjectGroupRequest> request = new HttpEntity<>(projectGroupRequest, headers);
+        HttpEntity<ProjectGroupRequest> requestEntity = new HttpEntity<>(projectGroupRequest, headers);
+            
+        ResponseEntity<ProjectGroupResponse> responseEntity = restTemplate.exchange(url, HttpMethod.POST, requestEntity,
+                ProjectGroupResponse.class);
+        return ResponseEntity.status(responseEntity.getStatusCode()).body(responseEntity.getBody());    }
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java:39]: Directly u can return the responseEntity

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java [hunk 15]
```diff
@@ -4,29 +4,53 @@
...
+	}
+
+	@PostMapping
+	public ResponseEntity<ProjectGroupResponse> createProjectGroup(@RequestBody ProjectGroupRequest projectGroupRequest) {
+	    return projectGroupService.createProjectGroup(projectGroupRequest);
+	}
+
+	@GetMapping("/{uid}")
+	public ResponseEntity<List<ProjectGroupResponse>> getProjectGroupsByProjectUid(@PathVariable Long uid) {
+	    List<ProjectGroupResponse> projectGroups = (List<ProjectGroupResponse>) projectGroupService.findByProjectUid(uid);
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:40]: Why is it typecasting? The return type in service class is already ResponseEntity.

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java [hunk 16]
```diff
@@ -4,29 +4,53 @@
...
+	public ProjectGroupController(ProjectGroupService projectGroupService) {
+	    this.projectGroupService = projectGroupService;
+	}
+
+	@PostMapping
+	public ResponseEntity<ProjectGroupResponse> createProjectGroup(@RequestBody ProjectGroupRequest projectGroupRequest) {
+	    return projectGroupService.createProjectGroup(projectGroupRequest);
+	}
+
+	@GetMapping("/{uid}")
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:38]: Rename it to projectId and similarly change in Path variable as well.

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java [hunk 17]
```diff
@@ -4,29 +4,53 @@
...
+	public ResponseEntity<List<ProjectGroupResponse>> getProjectGroupsByProjectUid(@PathVariable Long uid) {
+	    List<ProjectGroupResponse> projectGroups = (List<ProjectGroupResponse>) projectGroupService.findByProjectUid(uid);
+	    return ResponseEntity.ok(projectGroups);
+	}
+
+
+	@PutMapping()
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:45]: Add path variable groupUid

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java [hunk 18]
```diff
@@ -4,29 +4,53 @@
...
-        this.projectGroupService = projectGroupService;
-    }
+	}
 
-    @PostMapping
-    public ResponseEntity<ProjectGroupResponse> createProjectGroup(@RequestBody ProjectGroupRequest projectGroupRequest) {
-        return projectGroupService.createProGroup(projectGroupRequest);
-    }
-   }
+	@DeleteMapping
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:52]: Add Path variable groupUid

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java [hunk 19]
```diff
@@ -19,24 +23,72 @@ public class ProjectGroupService {
...
+
+public ResponseEntity<ProjectGroupResponse> findByProjectUid(Long id) {
+    	
+    	String url = configServiceURL + "/project-groups/" + id;
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java:43]: URL should be /project-groups/project/{projectId}

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 16 above)
Comment (shijukbabu-ust) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:38]: URL should be /project/{projectId}

**Review by shijukbabu-ust — CHANGES_REQUESTED** [review]

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 1 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java (hunk 3 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 4 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:22]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 5 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 7 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 10 above)
[the remark this answers is not in the document]
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 11 above)
[the remark this answers is not in the document]
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java (hunk 8 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java (hunk 13 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java (hunk 14 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 15 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 17 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java (hunk 18 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java (hunk 19 above)
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/service/ProjectGroupService.java]: resolved

**Review by shijukbabu-ust — CHANGES_REQUESTED** [review]

**Comment by shijukbabu-ust** [comment]
Add get mapping for GroupUID as well.

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java [hunk 20]
```diff
@@ -25,8 +30,24 @@ public ProjectGroupController(ProjectGroupService projectGroupService) {
         this.projectGroupService = projectGroupService;
     }
 
-    @PostMapping
+    @PostMapping("/project/{projectId}")
```
[the remark this answers is not in the document]
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java [hunk 21]
```diff
@@ -25,8 +30,24 @@ public ProjectGroupController(ProjectGroupService projectGroupService) {
...
     }
-   }
+
+    @GetMapping("/project/{projectId}")
+    public ResponseEntity<List<ProjectGroupResponse>> getProjectGroupsByProjectUid(@PathVariable Long uid) {
+        List<ProjectGroupResponse> projectGroups = (List<ProjectGroupResponse>) projectGroupService.findByProjectUid(uid);
+        return ResponseEntity.ok(projectGroups);
+    }
+
+    @PutMapping("/{uid}/{groupUid}")
```
[the remark this answers is not in the document]
Comment (Keerthi-212839) [reply src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java]: resolved

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java [hunk 22]
```diff
@@ -25,8 +30,24 @@ public ProjectGroupController(ProjectGroupService projectGroupService) {
...
 
-    @PostMapping
+    @PostMapping()
     public ResponseEntity<ProjectGroupResponse> createProjectGroup(@RequestBody ProjectGroupRequest projectGroupRequest) {
         return projectGroupService.createProGroup(projectGroupRequest);
     }
-   }
+
+    @GetMapping("/project/{projectId}")
+    public ResponseEntity<List<ProjectGroupResponse>> getProjectGroupsByProjectUid(@PathVariable Long uid) {
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:39]: Change variable name to `projectId` or Add as `@PathVariable("projectId")`

File: src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java [hunk 23]
```diff
@@ -25,8 +30,24 @@ public ProjectGroupController(ProjectGroupService projectGroupService) {
...
+        List<ProjectGroupResponse> projectGroups = (List<ProjectGroupResponse>) projectGroupService.findByProjectUid(uid);
+        return ResponseEntity.ok(projectGroups);
+    }
+
+    @PutMapping("/{groupUid}")
+    public ResponseEntity<ProjectGroupResponse> updateProjectGroup(@PathVariable Long uid, @PathVariable Long groupUid, @RequestBody ProjectGroupRequest projectGroupRequest) {
```
Comment (shijukbabu-ust) [inline src/main/java/com/ust/sourcecourse/orchestrator/controller/ProjectGroupController.java:45]: No need of uid variable

**Review by shijukbabu-ust — CHANGES_REQUESTED** [review]

**Review by shijukbabu-ust — APPROVED** [review]

**Pull request merged**